edx_rest_framework_extensions/auth/jwt/middleware.py

```diff
@@ -258,19 +261,56 @@ def process_view(self, request, view_func, view_args, view_kwargs):  # pylint: d
         # .. custom_attribute_name: has_jwt_cookie
         # .. custom_attribute_description: Enables us to see requests which have the full reconstituted
         #      JWT cookie. If this attribute is missing, it is assumed to be False.
-        monitoring.set_custom_attribute('has_jwt_cookie', has_reconstituted_jwt_cookie)
-
+        set_custom_attribute('has_jwt_cookie', has_reconstituted_jwt_cookie)
+
+        # DRF authentication does not set the request.user early enough for it to be used in process_request/
+        # process_view of middleware. This code enables JWT cookie authentication to set the request.user for
+        # middleware, before it will presumably happen again during DRF authentication.
+        # For more info on DRF and the request.user, see
+        # https://github.com/jpadilla/django-rest-framework-jwt/issues/45#issuecomment-74996698
+        #
+        # If the user has already been authenticated, we already have a user and intentionally avoid resetting it. As of
+        # Oct 2023, this would likely be due to session authentication in middleware. It is thus important for
+        # JwtAuthentication to verify that the session user and JWT user match. It is possible that this would be better
```

Comment:
@feanil: I also was also wondering about this case. In JwtAuthentication, if `ENABLE_SET_REQUEST_USER_FOR_JWT_COOKIE` **and** we get a JWT vs Session user mismatch, it might makes sense to error out, since we know it is possible that the middleware analytics and the JWT authentication may be for different users. Here we may want to fail even when JwtAuthentication succeeds, which we aren't yet doing for any other cases. Thoughts?

Reply:
**Update**: I added enforcement for this in a new commit that involved the least amount of refactoring I could think of.

Reply:
Sounds good, I agree that we should error out on mismatch here as well.